external_parser/unit_tests/test_example_joiner.cc

```diff
@@ -2,6 +2,58 @@
 #include "test_common.h"
 #include <boost/test/unit_test.hpp>
 
+BOOST_AUTO_TEST_CASE(example_joiner_test_ca) {
+  auto vw = VW::initialize("--quiet --binary_parser --cats 1 --min_value=0 --max_value=1", nullptr,
+                           false, nullptr, nullptr);
+
+  example_joiner joiner(vw);
+  v_array<example *> examples;
+  joiner.set_problem_type_config(v2::ProblemType_CA);
+  std::string input_files = get_test_files_location();
+  auto interaction_buffer = read_file(input_files + "/fb_events/ca_v2.fb");
```

Comment:
we should add to the `unit_test/test_files` README how these `ca_v2.fb` were generated, so that we keep track of everything